pandas/tests/io/parser/test_na_values.py

```diff
@@ -615,3 +616,45 @@ def test_nan_multi_index(all_parsers):
     )
 
     tm.assert_frame_equal(result, expected)
+
+
+@xfail_pyarrow
+def test_bool_and_nan_to_bool(all_parsers):
+    # GH#42808
+    parser = all_parsers
+    data = """0
+NaN
+True
+False
+"""
+    with pytest.raises(ValueError, match="NA values"):
+        parser.read_csv(StringIO(data), dtype="bool")
+
+
+def test_bool_and_nan_to_int(all_parsers):
+    # GH#42808
+    parser = all_parsers
+    data = """0
+NaN
+True
+False
+"""
+    if parser.engine == "pyarrow":
```

Comment:
shouldn't this xfail? e.g. is this a real error message

Reply:
No, this should fail. The error raised is different, because it comes from astype in case of Pyarrow

Reply:
For consistency of API, can we try catching this and re-raising?

Reply:
We could change the error. Controlling the message is difficult.

Raising a ValueError now

Reply:
Okay, at least standardizing the error class is good. 👍